src/renderers/shared/ReactDebugTool.js

```diff
@@ -240,15 +249,26 @@ var ReactDebugTool = {
     currentFlushNesting--;
     resumeCurrentLifeCycleTimer();
     emitEvent('onEndFlush');
+    if (isProfiling && hasAdvancedConsoleFeatures && currentFlushNesting === 0) {
+      console.groupEnd('React updates');
+    }
   },
   onBeginLifeCycleTimer(debugID, timerType) {
     checkDebugID(debugID);
     emitEvent('onBeginLifeCycleTimer', debugID, timerType);
+    if (isProfiling && hasAdvancedConsoleFeatures) {
+      var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+      var label = displayName + '#' + timerType;
+      console.time(label);
+    }
     beginLifeCycleTimer(debugID, timerType);
   },
   onEndLifeCycleTimer(debugID, timerType) {
     checkDebugID(debugID);
     endLifeCycleTimer(debugID, timerType);
+    var displayName = ReactComponentTreeHook.getDisplayName(debugID);
+    var label = displayName + '#' + timerType;
+    console.timeEnd(label);
```

Comment:
Or if two components are not named and you end up with 'Unknown' https://github.com/facebook/react/blob/master/src/isomorphic/hooks/ReactComponentTreeHook.js#L176